mcpwm_foc.c

```diff
@@ -2570,6 +2571,9 @@ void mcpwm_foc_adc_int_handler(void *p, uint32_t flags) {
 		motor_now->m_motor_state.iq = 0.0;
 		motor_now->m_motor_state.id_filter = 0.0;
 		motor_now->m_motor_state.iq_filter = 0.0;
+	#ifdef HW_HAS_INPUT_CURRENT_SENSOR
+		GET_INPUT_CURRENT_OFFSET();
```

Comment:
Should this really be done here? Just when the motor is released there might still be some current flowing through the body diodes due to the inductance, and that will offset this measurement. I think it is better to only sample the offset at boot.

Reply:
Hello Benjamin, Thanks for considering our pull request. 
That code only executes if a trigger was generated.  That trigger is generated inside this function **input_current_offset_measurement()**. 
The code as it is now, waits for a second after boot, and then takes some samples, and average them to get a better offset measurement. We did this because we got quite bad results if we took the very first samples of the ADC right from the beginning, so we used a delay of 1 second. It may be also possible to trigger offset measurements later, by calling the MEASURE_INPUT_CURRENT_OFFSET() macro, but now we only found it necessary at this point.